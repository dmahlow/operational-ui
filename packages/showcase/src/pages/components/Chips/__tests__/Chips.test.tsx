import * as React from "react"
import { render } from "enzyme"

import ChipsPage from "../Chips"

describe("Cards Page", () => {
  it("Should render correctly", () => {
    expect(render(<ChipsPage />)).toMatchSnapshot()
  })
})
