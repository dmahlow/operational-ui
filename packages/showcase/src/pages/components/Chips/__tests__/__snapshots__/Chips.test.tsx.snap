// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Page Should render correctly 1`] = `
<div>
  <div
    class="css-by30gj"
  >
    Chips
  </div>
  <p>
    Most commonly used for filters, these elements represent small bits of information that give a sense of context to the user. Chips can be interactive, or simply informative. They can take on any color passed through \`props\`, along with a symbol for the button that will be displayed if click behavior is detected.
  </p>
  <div
    style="display:flex;"
  >
    <div
      class="css-10834mi chip"
    >
      Chip 1
    </div>
    <div
      class="css-10834mi chip"
    >
      Chip 2
    </div>
  </div>
  <h4>
    Usage
  </h4>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          color
        </td>
        <td>
          What color of chip would you like? It can be a hex value or a named theme color.
        </td>
        <td>
          The \`primary\` color of your theme.
        </td>
        <td>
          string
        </td>
        <td>
          ✅
        </td>
      </tr>
      <tr>
        <td>
          symbol
        </td>
        <td>
          The symbol that is shown on mouse over of a clickable chip.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          ✅
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          Is this interactive? If yes, what happens when the chip is clicked? This is commonly used to delete a filter in a list of filters.
        </td>
        <td />
        <td>
          func
        </td>
        <td>
          ✅
        </td>
      </tr>
    </tbody>
  </table>
  <div
    style="margin-bottom:32px;"
  />
  <div
    class="css-by30gj"
  >
    PlusChip
  </div>
  <p>
    A PlusChip is a specialized chip that bears no label, but simply takes an action and displays a symbol prompting the action. These chips can either contain their own symbol, or simply have a plus if no symbol is specified.
  </p>
  <h4>
    Usage
  </h4>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          t]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-81lp1j"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          color
        </td>
        <td>
          What color of chip would you like? It can be a hex value or a named theme color.
        </td>
        <td>
          Black
        </td>
        <td>
          string
        </td>
        <td>
          ✅
        </td>
      </tr>
      <tr>
        <td>
          size
        </td>
        <td>
          The size of the chip in pixels. This can vary for obvious reasons.
        </td>
        <td>
          16
        </td>
        <td>
          number
        </td>
        <td>
          ✅
        </td>
      </tr>
      <tr>
        <td>
          onClick
        </td>
        <td>
          A function that will be called on click of the PlusChip.
        </td>
        <td />
        <td>
          func
        </td>
        <td>
          ✅
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
